File: Creational/Builder/JavaScript/carDirector.js
// carDirector.js

const CarBuilder = require("./carBuilder");

class CarDirector {
  buildLuxuryCar() {
    return new CarBuilder()
            .setEngine("V8 Turbo")
            .setGPS(true)
            .setSunroof(true)
            .setSeats("Leather")
            .build();
  }

  buildEconomyCar() {
    return new CarBuilder()
            .setEngine("1.6L Petrol")
            .setGPS(false)
            .setSunroof(false)
            .setSeats("Standard")
            .build();
  }
}

module.exports = CarDirector;

File: Creational/Builder/Python/car_director.py
# car_director.py

from car_builder import CarBuilder

class CarDirector:
  def build_luxury_car(self):
    return CarBuilder()\
            .set_engine("V8 Turbo")\
            .set_gps(True)\
            .set_sunroof(True)\
            .set_seats("Leather")\
            .build()

  def build_economy_car(self):
    return CarBuilder()\
            .set_engine("1.6L Petrol")\
            .set_gps(False)\
            .set_sunroof(False)\
            .set_seats("Standard")\
            .build()
